packages/shared-ui/src/components/canvas/context/FlowGraphContext.tsx

```
@@ -729,6 +686,12 @@ export function FlowGraphProvider({ children }: IFlowGraphProviderProps): ReactE
 	const addNode = useCallback(
 		(data: INodeData, position?: { x: number; y: number }, type: INodeType = INodeType.Default): string => {
 			if (isLocked) return '';
+			// Log connection data on existing nodes to verify it's preserved
+			for (const n of nodes) {
+				const nd = n.data as any;
+				if (nd.input?.length || nd.control?.length) {
+				}
```

ESLint

[error] Empty block statement.

(no-empty)